assets/wizards/advertising/index.js

```diff
@@ -258,10 +259,11 @@ class AdvertisingWizard extends Component {
 	prepareData = data => {
 		return {
 			...data,
-			adUnits: data.ad_units.reduce( ( result, value ) => {
-				result[ value.id ] = value;
-				return result;
-			}, {} ),
+			adUnits:
+				data?.ad_units?.reduce( ( result, value ) => {
```

Comment:
This would be in the case of `get_ad_units()` returning error due to failed sync on `newspack-ads`. This was actually not necessary because the `prepareData()` method would never be called in this situation.

a5df3f3e5f7c1f3c54639082d64a39fa6ab7240f reverts this change and also fix error handling when `get_ad_units()` returns WP_Error instead of failing with exception, as it is handled currently.

Reply:
Update: I left some residual code, 9b74997810afd433b652e1ff58d955262854f8c1 reverts it completely.